server/src/api/modules/groups/group.utils.ts

```diff
@@ -16,4 +19,20 @@ function buildDefaultSocialLinks() {
   };
 }
 
-export { sanitizeName, buildDefaultSocialLinks };
+function sortMembers(group: Omit<GroupDetails, 'socialLinks' | 'memberCount'>): MembershipWithPlayer[] {
+  if (group.roleOrders && group.roleOrders.length) {
+    // this assumes roleOrders is sorted by index ascending out of the database
+    return [...group.memberships].sort(
+      (a, b) =>
+        (group.roleOrders?.find(order => order.role === a.role)?.index ?? 0) -
```

Comment:
Maybe we could create a map from `group.roleOrders` and then use it for O(1) lookups? Otherwise you'll be iterating that array a lot of times, especially for large groups.

something like

```typescript
const roleOrderMap = new Map(group.roleOrders.map(r => [r.role, r.index]));

return [...group.memberships].sort((a, b) => {
  return (roleOrderMap.get(a.role) ?? 0) - (roleOrderMap.get(b.role) ?? 0) // should these default to 0?
})
```

Also note my code comment above ^ 🤔